fix: importer being a virtual module

brillout opened the pull request:
When the importer is a virtual module, there is no current directory and vite-plugin-glob fails. This PR fixes it.

antfu commented on src/transform.ts:
```diff
@@ -19,7 +19,7 @@ export async function transform(
   id = toPosixPath(id)
   root = toPosixPath(root)
-  const dir = dirname(id)
+  const dir = !id.includes('/') ? null : dirname(id)
```
This might not be a safe assumption to me, a version module could still have `/` like `virtual:some/thing`.

How does Vite core handle that?

brillout replied:
👍 Done https://github.com/antfu/vite-plugin-glob/pull/18/commits/1d4202e50423cc9d8463fb67693dc732b209f831.

antfu commented on src/transform.ts:
```diff
@@ -68,6 +68,13 @@ export async function transform(
+          if (!filePath.startsWith('.'))
+            filePath = `/${filePath}`
+          return { filePath, importPath: filePath }
+        }
```
I guess maybe we should require a virtual module to always glob from an absolute path

antfu commented:
Could we add a few tests for it? Thanks!

brillout commented:
Done.

Added tests in https://github.com/antfu/vite-plugin-glob/pull/18/commits/4d5d484b3a25ada69f7d10b79b25138f6830357e.

Also fixed a bug in https://github.com/antfu/vite-plugin-glob/pull/18/commits/10205403b749a5bae1407a13ab37c9ad6d855d84.

The tests already bear fruit :-).
